tests/rdma/test_pfc_pause_lossless.py

```diff
@@ -0,0 +1,328 @@
+                                  gw_incr_step='0.0.0.0')
+    start_protocols(session)
+
+    test_traffic = create_ipv4_traffic(session=session,
+                                       name='Test Data Traffic',
+                                       source=topo_sender,
+                                       destination=topo_receiver,
+                                       pkt_size=DATA_PKT_SIZE,
+                                       duration=exp_dur,
+                                       rate_percent=50,
+                                       start_delay=start_delay,
+                                       dscp_list=test_dscp_list,
+                                       lossless_prio_list=test_prio_list)
+
+    bg_priority_list = [b for b in range(8) if b not in test_prio_list]
+    background_traffic = create_ipv4_traffic(session=session,
+                                             name='Background Data Traffic',
+                                             source=topo_sender,
+                                             destination=topo_receiver,
+                                             pkt_size=DATA_PKT_SIZE,
+                                             duration=exp_dur,
+                                             rate_percent=50,
+                                             start_delay=start_delay,
+                                             dscp_list=bg_dscp_list,
+                                             lossless_prio_list=bg_priority_list)
+
+    # Pause time duration (in second) for each PFC pause frame.
+    pause_dur_per_pkt = 65535 * 64 * 8.0 / (port_bw * 1000000)
+
+    # Do not specify duration here as we want it keep running.
+    if send_pause_frame:
+        pfc_traffic = create_pause_traffic(session=session,
+                                           name='PFC Pause Storm',
+                                           source=rx_port,
+                                           pkt_per_sec=1.1/pause_dur_per_pkt,
+                                           start_delay=0,
+                                           global_pause=False,
+                                           pause_prio_list=test_prio_list)
+
+    start_traffic(session)
+
+    # Wait for test and background traffic to finish.
+    time.sleep(exp_dur + start_delay + 1) 
+
+    # Capture traffic statistics.
+    flow_statistics = get_traffic_statistics(session)
+    logger.info(flow_statistics)
+
+    for row_number, flow_stat in enumerate(flow_statistics.Rows):
+        tx_frames = int(flow_stat['Tx Frames'])
+        rx_frames = int(flow_stat['Rx Frames'])
+
+        if 'Test' in flow_stat['Traffic Item']:
+            if test_traffic_pause_expected:
+                pytest_assert(tx_frames > 0 and rx_frames == 0,
+                    "Test traffic should be fully paused")
+            else:
+                pytest_assert(tx_frames > 0 and tx_frames == rx_frames,
+                    "Test traffic should not be impacted")
```

Comment:
Can you also check if tx_frames * PKT_SIZE is close to rate_percent (50/100) * duration * link capacity?

Reply:
An equivalent check is implemented.